File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> hm = new HashMap<>();
        
        for(int i=0; i<participant.length; i++){
            hm.put(participant[i], hm.getOrDefault(participant[i], 0) + 1);
        }
        
        for(int i=0; i<completion.length; i++){
            hm.put(completion[i], hm.getOrDefault(completion[i], 0) - 1);
            if(hm.get(completion[i]) <= 0) hm.remove(completion[i]);
        }
        
        String answer = "";
        
        for(String key : hm.keySet()){
            answer = key;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    parD = {}
    
    for i in range(len(participant)):
        parD[participant[i]] = parD.get(participant[i], 0) + 1
        
    for i in range(len(completion)):
        parD[completion[i]] -= 1
    
    for name in parD:
        if parD[name] > 0:
            return name